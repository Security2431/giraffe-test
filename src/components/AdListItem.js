import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Link } from 'react-router-dom';

const theme = {
  Card: {
    marginBottom: '30px',
  },
};

class AdListItem extends React.Component {
  static propTypes = {
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author_name: PropTypes.string.isRequired,
    created_at_datetime: PropTypes.string.isRequired,
    activeUser: PropTypes.string.isRequired,
    descrFixedLength: PropTypes.number.isRequired,
  };

  handleDeleteClick = (event, id) => {
    event.preventDefault();

    // eslint-disable-next-line
    this.props.deleteAd(id);
  };

  sliceText = (text, textLength) => {
    if (!textLength) {
      return text;
    }

    return text && text.length > textLength ? `${text.slice(0, textLength)}\xa0...` : text;
  };

  render() {
    const {
      _id: id,
      title,
      description,
      author_name: author,
      created_at_datetime: date,
      descrFixedLength,
      activeUser,
    } = this.props;

    const datetime = moment(date).format('DD/MM/YYYY');
    const isAdFromMe = activeUser === author;

    return (
      <div className="card-body" style={theme.Card}>
        <h5 className="card-title">
          <Link to={`${id}`}>{title}</Link>
        </h5>
        <p className="d-flex justify-content-beetween no-gutters card-text">
          <small className="text-muted col-sm-6">{author}</small>
          <small className="text-muted text-right col-sm-6">{datetime}</small>
        </p>
        <p className="card-text">{this.sliceText(description, descrFixedLength)}</p>
        {isAdFromMe && (
          <div className="btn-group d-flex">
            <Link className="btn btn-dark" to={`/edit/${id}`}>
              Edit
            </Link>
            <button
              className="btn btn-warning"
              type="button"
              onClick={e => this.handleDeleteClick(e, id)}
            >
              Delete
            </button>
          </div>
        )}
      </div>
    );
  }
}

export default AdListItem;
