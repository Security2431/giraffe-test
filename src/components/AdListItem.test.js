/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AdListItem from './AdListItem';

// Return a fixed timestamp when moment().fromNow() is called
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const mockProps = {
  _id: '123',
  title: 'Title',
  description: 'Hello, World!',
  author_name: 'Admin',
  created_at_datetime: '2019-05-03T12:59:23+03:00',
  activeUser: 'Admin',
  descrFixedLength: 100,
};

describe('<AdListItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <AdListItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <AdListItem {...mockProps} />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
