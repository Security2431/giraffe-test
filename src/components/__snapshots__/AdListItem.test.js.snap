// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdListItem /> renders correctly 1`] = `
<div
  className="card-body"
  style={
    Object {
      "marginBottom": "30px",
    }
  }
>
  <h5
    className="card-title"
  >
    <a
      href="/123"
      onClick={[Function]}
    >
      Title
    </a>
  </h5>
  <p
    className="d-flex justify-content-beetween no-gutters card-text"
  >
    <small
      className="text-muted col-sm-6"
    >
      Admin
    </small>
    <small
      className="text-muted text-right col-sm-6"
    >
      2 days ago
    </small>
  </p>
  <p
    className="card-text"
  >
    Hello, World!
  </p>
  <div
    className="btn-group d-flex"
  >
    <a
      className="btn btn-dark"
      href="/edit/123"
      onClick={[Function]}
    >
      Edit
    </a>
    <button
      className="btn btn-warning"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
